### DISCUSSION TITLE
Cant find the sensor for power usage.

### DISCUSSION THREAD

**Discussion by lynxus** [hultenvp/solis-sensor#327] [General]
Hi Guys,

What is the entity / sensor that does the power usage? This must have been asked before, but i cant see it..
I cant find it anywhere!

Within the python test script, it can use:
totalHousePowerUsage = (json['data']['totalLoadPower'])

But I cant find it in Home Assistant?

Any ideas?
Thanks
Graham

![image](https://github.com/hultenvp/solis-sensor/assets/6402873/693d0c8d-d113-4606-83e5-0fdfc8caa153)

**Comment by artarus** [comment]
> Hi Guys,
> 
> What is the entity / sensor that does the power usage? This must have been asked before, but i cant see it.. I cant find it anywhere!
> 
> Within the python test script, it can use: totalHousePowerUsage = (json['data']['totalLoadPower'])
> 
> But I cant find it in Home Assistant?
> 
> Any ideas? Thanks Graham
> 
> ![image](https://user-images.githubusercontent.com/6402873/270156014-693d0c8d-d113-4606-83e5-0fdfc8caa153.png)

Hi Graham,
There no entity for load power. But it can be easy added code. A key for this is 'familyLoadPower' from station detail API. If you need i have made this changes in the files for self and it is working well.
![load power](https://github.com/hultenvp/solis-sensor/assets/26852635/326ead54-f381-48f4-a7ad-f577abba3b0f)

**Reply by lynxus** [reply]
ok thanks ill take a look.

**Reply by lynxus** [reply]
Yeah if you can share the code, that would be useful.

**Reply by artarus** [reply]
Just replace the files in custom_components/solis and restart HA
[solis.zip](https://github.com/hultenvp/solis-sensor/files/12709566/solis.zip)

**Reply by lynxus** [reply]
thanks, ill give it a shot :)

**Reply by lynxus** [reply]
would be good to see why @hultenvp hasn't included this in the normal build?

**Reply by hultenvp** [reply]
This integration evolved from ginlog.com which was replaced by soliscloud.com, which has more detailed data to share. Adding the new data happened a bit on a on-request basis.
I still plan a big update that will expose all available Soliscloud data, but I fail to find the time to finish the work I started. Perhaps this winter.
For this particular sensor if @artarus or you could create a pull request for the changes then I'm happy to integrate and create a new release.

Cheers

**Comment by ccj2740** [comment]
Hi first of all nice work all...

@artarus I am looking for the same sensor in version 3.5.1.
I wonder if it would it be possible to add the "Solis family load sensor" to @hultenvp 3.5.1 version as the current "Solis Total Consumption power sensor" have issues with delays.  ?

**Comment by ccj2740** [comment]
@artarus I tried to add your files solis.zip and overwrite the current files from 3.5.1.
I got the "Solis family load sensor" , but it is delayed so the data refresh is like 5min old and then it doesn´t match my Solis Power Grid total power which is updating every 1min. 

Would it be possible to get it working same way as they look in soliscloud:
![image](https://github.com/hultenvp/solis-sensor/assets/126421743/2356b8fa-5ed5-42ba-a38b-0126e3c4e346)

**Reply by artarus** [reply]
@ccj2740 @hultenvp
[fix.zip](https://github.com/hultenvp/solis-sensor/files/13957657/fix.zip)
 
Hi, unfortunately i do not have access to my multi inverter plant right now, i will investigate the problem after 3 weeks. Temporarily you can try and use this fix.

**Reply by ccj2740** [reply]
Thanks @artarus. I tried the fix.zip and now the "Solis Total Consumption power sensor"  is updating correctly again...

BUT the "Solis Total Consumption power sensor" is diffrent data than the data in Solis cloud:
![image](https://github.com/hultenvp/solis-sensor/assets/126421743/5bbe2bdc-2f03-4a45-a813-3c58aa5c4a00)

Example when I compare from Solis cloud to the sensor in home assistant, then I see a difference of around 200-300W...

I think the reason for this difference is that it is taking:
Total Consumption Power - Backup Load Power 2,71 - 0,202 =
![image](https://github.com/hultenvp/solis-sensor/assets/126421743/5b0f4605-ba43-43df-bd66-ccd76464dfe4)


I don´t think it should include the "backup load power" as it gives me wrong information ?

Also I am Happy to hear that you want to investigate it after 3 weeks I appreciate that very much :)

**Reply by artarus** [reply]
@ccj2740 I don't have batteries in my system, so i can not answer to your question how solis calculates it. Also in the fix i have added "Solis Plant Total Consumption power" it is same sensor as "Solis Total Consumption power sensor" in the version 3.5.1. Maybe issue that SolisCloud portal update rate is not same as HA? Because i see my SolisCloud portal updates every 5 min. I tested in my single inverter system: All values are correct: "Solis Total Consumption power sensor" same as in Cloud inverter details. "Solis Plant Total Consumption power" same as in plant overview. And why power values is important for you instead an energy?

**Reply by ccj2740** [reply]
@artarus : Yes I noticed that new sensor and also tried it , but that one have delay with 5min , BUT you have fixed it so the Solis Total Consumption power is now working for me :). Only issues I still have is that it for some reason take my "Solis Total Consumption power and - Solis Backup Load Power" I am not sure why it does that since when I check in Solis cloud it is not.

Not sure what you mean with : And why power values is important for you instead an energy? 
My total power usage in house is what I am trying to be correct , so if you have some other solution to show it correct I would very very happy :)

EDIT:
I got help to create a template sensor that take:
state: "{{states('sensor.solis_total_consumption_power') | int + states('sensor.solis_backup_load_power') | int }}"

So now my consumption power is the same as shown in Solis cloud.. Thanks to you all!!!

**Reply by hultenvp** [reply]
Thanks @artarus : Integrated your changes from fix.zip

**Comment by jbtibor** [comment]
Hi, what is familyLoadPower? The value returned by API doesn't match any value on the Solis cloud summary page. I'm comparing the correct timestamps.
